### DISCUSSION TITLE
GMTSAR SBAS OUTPUT-Displacement maps

### DISCUSSION THREAD

**Discussion by AnuradhaSL** [gmtsar/gmtsar#821] [Q&A]
I am currently working with GMTSAR for my doctoral research, and I'm encountering some confusion regarding the master image in each displacement map generated as SBAS outputs.

In the output files, the names of the displacements are identical to the names of the scenes. However, I've noticed that the first displacement map, which shares the same name as the first scene, consistently shows a value of zero. I'm wondering if this represents the baseline for the other displacement maps. Is it essentially the super master image?

**Comment by Xiaohua-Eric-Xu** [comment]
Yes the first grid is always zeros, since that is the start of your time-series. However, it may not be your supermaster image. As e.g., you may coregister your data stack to a specific date that's not the start date of your time-series.

**Reply by anuradhacssteap** [reply]
Dear Eric,
Thank you so much for this reply. It's very helpful.
Now I understood the difference between super master image and first scene of the time series. I kindly required one more clarification. 
Hence the first grid is zero, do other grids show displacements with respect to the first grid?

**Reply by Xiaohua-Eric-Xu** [reply]
Yes they are displacements with respect to the first grid (sorry for the late response).

**Reply by anuradhacssteap** [reply]
Dear Prof. Eric,
Thank you so much for your response. Now, I am very clear about the time series and SBAS results.
I have one more question, but not directly about SBAS results. GMTSAR is a great package; I currently use it for my doctoral project. 
1) When the number of interferograms exceeds 250, I mean when the number of interferograms is around 500, SBAS processing is killed. I have checked your previous explanations and understood it's due to insufficient memory (RAM). I am using the "-mmap" option right now, and it's going on for more than 5hrs now. 
2) I would kindly like to know that GMTSAR has some limitations because of the large number of interferogram processes. This is really important to me as I have to process lots of data with GMTSAR. 
Also, I want to kindly know how can I apply "**gmt grdcut unwrap.grd -Rmin/xmax/ymin/ymax -Gnew_unwrap.grd**" for all unwrap folders at one time or paralley. 

Thank you.

**Reply by Xiaohua-Eric-Xu** [reply]
Hi @anuradhacssteap, the -mmap will use disk space to swap as memory. It may make things slower depending on how fast your disk speed is. I would try to compile sbas_parallel.c to use multi cores for solving the time-series. 
For processing lots of data, we've included some scripts that uses GNU parallel to spread the jobs into multiple threads, such as intf_tops_parallel.csh or unwrap_parallel.csh. Let me know if you would like to have other parts paralleled. 
For a for loop, you could easily implement that with C-Shell as
```
foreach line (`ls -d 20*`)
  cd $line
  gmt ...
  cd ..
end
```
